public/js/app.js

```diff
@@ -132,6 +132,13 @@ const getQueryParam = (name) => {
   return urlParams.get(name);
 };
 
+const setQueryParam = (name, value) => {
+  let urlParams = new URLSearchParams(window.location.search);
+  urlParams.set(name, value);
+  urlParams.sort();
```

Comment:
why is this necessary?

Reply:
So that our URL is always the same. It's better for CDN caching and analytics.